sensor_control/battery_monitor/nodes/battery_monitor.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+
+'''
+Battery Monitor: A simple script that approximates the battery voltage by
+averaging the supply voltage to each of the four thrusters.
+'''
+
+
+from roboteq_msgs.msg import Feedback
+import rospy
+from std_msgs.msg import Float32
+from sub8_alarm import AlarmBroadcaster
+
+
+__author__ = "Anthony Olive"
+__maintainer__ = "Anthony Olive"
+__email__ = "anthony@iris-systems.net"
+__copyright__ = "Copyright 2016, MIL"
+__license__ = "MIT"
+
+
+# Voltage threshold for the initial low voltage alarm
+# 22.1V (approximately 30%) is the recommended value
+battery_low = 22.1
+
+# Voltage threshold for the critically low voltage alarm
+# 20.6V (approximately 15%) is the recommended value
+battery_critical = 20.6
+
+# Voltage threshold for the total system kill alarm
+# 19.5V (approximately 5%) is the recommended value
+battery_kill = 19.5
+
+rospy.init_node('battery_monitor')
+
+
+class BatteryMonitor():
+
+    def __init__(self):
+
+        # Initialize the four receiving variables to max voltage
+        self.supply_voltage = [29.4, 29.4, 29.4, 29.4]
+
+        # Initialize the average voltage to max
+        self.voltage = 29.4
+
+        # Initialize the variables for handling averaging (in an attempt to eliminate outliers)
+        self.avg_count = [0, 0, 0, 0]
+        self.avg_value = [0, 0, 0, 0]
+
+        # The publisher for the averaged voltage
+        self.pub_voltage = rospy.Publisher("/battery_monitor", Float32, queue_size=1)
+
+        # Subscrives to the feedback from each of the four thrusters
+        rospy.Subscriber("/FL_motor/feedback", Feedback, self.add_to_fl)
```

Comment:
Was there a reason to keep all four averages separately vs having all the callbacks the same and keeping track of one list?